calculator.go

```diff
@@ -3,30 +3,100 @@ package calculator
 
 import (
 	"errors"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
 )
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-	return a + b
+// Add takes two or more numbers and returns the result of adding them together.
+func Add(a, b float64, nums ...float64) float64 {
+	result := a + b
+	for _, n := range nums {
+		result += n
+	}
+	return result
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
+// Subtract takes two or more numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return a - b
+func Subtract(a, b float64, nums ...float64) float64 {
+	result := a - b
+	for _, n := range nums {
+		result -= n
+	}
+	return result
 }
 
-// Multiply takes two numbers and returns the result of multiplication of both
+// Multiply takes two or more numbers and returns the result of multiplication of both
 // the numbers
-func Multiply(a, b float64) float64 {
-	return a * b
+func Multiply(a, b float64, nums ...float64) float64 {
+	result := a * b
+	for _, n := range nums {
+		result *= n
+	}
+	return result
 }
 
-// Divide takes two numbers and returns the result of dividing the first number
-// by second number. It returns error if the division operation is not valid
-func Divide(a, b float64) (float64, error) {
+// Divide takes two or more numbers and returns the result of dividing the given numbers.
+// It returns error if the division operation is not valid
+func Divide(a, b float64, nums ...float64) (float64, error) {
 	if b == 0 {
 		return 0, errors.New("Divide by Zero")
 	}
-	return a / b, nil
+	result := a / b
+
+	for _, n := range nums {
+		if n == 0 {
+			return 0, errors.New("Divide by Zero")
+		}
+		result /= n
+	}
+	return result, nil
+}
+
+// SquareRoot returns the square root of the provided number , error if the input
+// number is invalid
+func SquareRoot(a float64) (float64, error) {
+	if a < 0 {
+		return 0, fmt.Errorf("bad input %f; square root is undefined for negative values", a)
+	}
+	return math.Sqrt(a), nil
+}
+
+// ComputeString processes the input string as two operands separated by the operator
+// and returns the result of that operation. Returns error if the string is not in
+// the acceptable form.
+func ComputeString(str string) (float64, error) {
+
+	opIndex := strings.IndexAny(str, "+-*/")
+	if opIndex == -1 {
+		return 0, errors.New("Invalid string")
+	}
+
+	operand1 := strings.TrimSpace(str[0:opIndex])
```

Comment:
How about splitting the input string on spaces? Then we should have a three-element slice: operand, operator, operand.

Reply:
But then we also want to support expressions without space, right ?

Reply:
Yes, that's a good point. You could use the `text/scanner` package to tokenize the input, perhaps. (That's probably overkill for something like this, but it might be fun to try.)